src/celeritas/track/SortTracksAction.cc

```diff
@@ -91,6 +97,7 @@ std::string SortTracksAction::label() const
 void SortTracksAction::execute(CoreParams const&, CoreStateHost& state) const
 {
     detail::sort_tracks(state.ref(), track_order_);
+    state.count_tracks_per_action(track_order_);
```

Comment:
I feel like doing that would leak the implementation of `CoreState` because `detail::count_tracks_per_action` needs reference to the final host collection where to copy results and the intermediate host/device collection  in which to do the counting. These two collections are member variables of `CoreState`. We'd have to update the caller if we change anything